web_infrastructure/jenkins_job.py

```diff
@@ -0,0 +1,326 @@
+description:
+    - Manage Jenkins jobs by using Jenkins REST API
+requirements:
+  - "python-jenkins >= 0.4.12"
+  - "lxml >= 3.3.3"
+version_added: "2.2"
+author: "Sergio Millan Rodriguez"
+options:
+  config_file:
+    description:
+      - Absolute path to the config.xml for an specific job.
+    required: true
+  name:
+    description:
+      - Name of the Jenkins job.
+    required: true
+  password:
+    description:
+      - Password to authenticate with the Jenkins server.
+    required: false
+  state:
+    description:
+      - Action to take with the Jenkins job.
+    required: false
+    choices: ['present', 'absent', 'disabled']
+    default: present
+  token:
+    description:
+      - API token used to authenticate alternatively to password.
+    required: false
+  url:
+    description:
+      - Url where the Jenkins server is accessible.
+    required: false
+    default: http://localhost:8080
+  user:
+    description:
+       - User to authenticate with the Jenkins server.
+    required: false
+'''
+
+EXAMPLES = '''
+# Create a jenkins job using basic authentication
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# Create a jenkins job using the token
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    token: asdfasfasfasdfasdfadfasfasdfasdfc
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# Delete a jenkins job using basic authentication
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: absent
+    url: "http://localhost:8080"
+    user: admin
+
+# Delete a jenkins job using the token
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    token: asdfasfasfasdfasdfadfasfasdfasdfc
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# Disable a jenkins job using basic authentication
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: disabled
+    url: "http://localhost:8080"
+    user: admin
+
+# Disable a jenkins job using the token
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    token: asdfasfasfasdfasdfadfasfasdfasdfc
+    state: disabled
+    url: "http://localhost:8080"
+    user: admin
+
+# To re-enable a job just make it be present
+- jenkins_job:
+    config_file: /path/to/config.xml
+    name: test
+    password: admin
+    state: present
+    url: "http://localhost:8080"
+    user: admin
+
+# A production ready example
+- template:
+    src: test-job.xml.j2
+    dest: /tmp/test-job-config.xml
+    mode: 0644
+    owner: jenkins
+    group: jenkins
+
+- jenkins_job:
+    config_file: /tmp/test-job-config.xml
+    name: test-job
+    token: abcdefghijklmnopqrstuvwxyz
+    state: present
+    url: "https://jenkins.mydomain.com"
+    user: my_user
+'''
+
+RETURN = '''
```

Comment:
As the code returns results, these must be documented here (but do not document the common return `changed`:

~~~
 module.exit_json(changed=changed, name=self.name, state=self.state, url=self.jenkins_url)
~~~